### PR TITLE
Issue 10386: Reworked match

### LINKED ISSUE #10386: Searching for similar interests brings php-warning
- [ ] I have searched open and closed issues for duplicates

### Bug Description

Searching for "similar interests" brings php-warning. See screenshot at the end.

### Steps to Reproduce

1.  got to "contacts" page
2.  klickt on "find similar interests" 
3.  see screenshot

Actual Result:

see screenshot in description above

Expected Result:

no warnings

### Screenshots

grafik

### Platform Info

Friendica Version:

2021.06-rc commit bb60ed409

Friendica Source: git

PHP version: 7.3

SQL version: mariadb 10.3

### PR DESCRIPTION
**Pull request by annando**
Fixes #10386

The "match" mechanism had some issues, they are fixed now.

### CODE REVIEW

File: mod/match.php [hunk 1]
```diff
@@ -123,9 +96,37 @@ function match_content(App $a)
 	$tpl = Renderer::getMarkupTemplate('viewcontact_template.tpl');
 	$o = Renderer::replaceMacros($tpl, [
 		'$title'    => DI::l10n()->t('Profile Match'),
-		'$contacts' => $entries,
-		'$paginate' => $paginate
```
Comment (MrPetovan) [inline mod/match.php:127]: Please remove the unused template variable in the template file as well.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: mod/match.php (hunk 1 above)
Comment (annando) [reply mod/match.php:127]: The template is used in several places, with and without pagination.

Without:
- https://github.com/friendica/friendica/blob/2021.06-rc/mod/suggest.php#L52
- https://github.com/friendica/friendica/blob/2021.06-rc/src/Module/Conversation/Network.php#L172

With:
- https://github.com/friendica/friendica/blob/2021.06-rc/src/Module/BaseSearch.php#L133

Comment (MrPetovan) [reply mod/match.php:127]: Got it.

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
Please resolve the messages.po merge conflict.

**Comment by annando** [comment]
@MrPetovan done

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**